pkg/components/storage.go

```diff
@@ -121,25 +157,168 @@ func startCSIPlugin(ctx context.Context, cfg *config.Config, kubeconfigPath stri
 		klog.Warningf("Failed to apply clusterrolebinding %v: %v", crb, err)
 		return err
 	}
-	if err := assets.ApplyConfigMapWithData(ctx, cm, map[string]string{"lvmd.yaml": lvmdRenderParams["lvmd"].(string)}, kubeconfigPath); err != nil {
+	if err := assets.ApplyConfigMaps(ctx, []string{cm_ver}, nil, nil, kubeconfigPath); err != nil {
 		klog.Warningf("Failed to apply configMap %v: %v", crb, err)
 		return err
 	}
-	if err := assets.ApplyConfigMaps(ctx, []string{cm_ver}, nil, nil, kubeconfigPath); err != nil {
-		klog.Warningf("Failed to apply configMap %v: %v", crb, err)
+	if err := assets.ApplyServices(ctx, svc, nil, map[string]interface{}{}, kubeconfigPath); err != nil {
+		klog.Warningf("Failed to apply service %v: %v", svc, err)
 		return err
 	}
 	if err := assets.ApplyDeployments(ctx, deploy, renderTemplate, renderParamsFromConfig(cfg, nil), kubeconfigPath); err != nil {
 		klog.Warningf("Failed to apply deployment %v: %v", deploy, err)
 		return err
 	}
-	if err := assets.ApplyDaemonSets(ctx, ds, renderTemplate, renderParamsFromConfig(cfg, lvmdRenderParams), kubeconfigPath); err != nil {
-		klog.Warningf("Failed to apply daemonsets %v: %v", ds, err)
+	if err := assets.ApplyValidatingWebhookConfiguration(ctx, vwc, kubeconfigPath); err != nil {
+		klog.Warningf("Failed to apply validating webhook configuration %v: %v", vwc, err)
 		return err
 	}
-	if err := assets.ApplySCCs(ctx, scc, nil, nil, kubeconfigPath); err != nil {
-		klog.Warningf("Failed to apply sccs %v: %v", scc, err)
+	if err := assets.ApplyGeneric(ctx, lvmClusters, nil, map[string]interface{}{}, nil, kubeconfigPath); err != nil {
+		klog.Warningf("Failed to apply lvmcluster: %v", err)
 		return err
 	}
+
 	return nil
 }
+
+// loadCSIPluginConfig sets up a file watcher on the configuration directory. It first creates a cancellable context
+// and determines the directory and file path of the configuration file. It then checks if the configuration directory
+// and file exist. If the configuration file exists, it copies the user's configuration, otherwise, it copies the default configuration.
+//
+// The function then creates a new file watcher and adds the configuration directory to the watcher's watch list.
+// It listens for file events in a separate goroutine. If the configuration file is modified, it copies the user's configuration again.
+// If the configuration file is removed, it copies the default configuration. If the file permissions are changed, it logs a warning.
+//
+// The function is designed to ensure that lvms always uses the most recent configuration through its hot-reload mechanism,
+// whether it's user-defined or the default, and reacts to changes in the configuration file in real-time.
+func loadCSIPluginConfig(ctx context.Context,
+	// usrCfg is the user's configuration file.
+	usrCfg string,
+	runtimeCfg string,
+) (*lvmd.Lvmd, error) {
+	usrCfgDir := filepath.Dir(usrCfg)
+	ctx, cancel := context.WithCancelCause(ctx)
```

Comment:
Just to be sure. AFAIK the context cancelation doesn't propagate back to parents, right?
So, if we get some error while watching, it won't be propagated back to main threat

Reply:
thats right, cancel will only cancel the context return from context.WithCancelCause, not the parent ctx. Ive tried this and tested around and microshift ctx keeps on running